### PR TITLE
Add device ID to user model

### PR DESCRIPTION
**Pull request by kaapstorm**
Fixes PR #14087

@czue @esoergel @NoahCarnahan cc @proteusvacuum @sravfeyn @biyeun

### CODE REVIEW

File: corehq/apps/users/models.py
Comment (proteusvacuum) [inline corehq/apps/users/models.py]: how come you aren't using `DeviceIdLastUsed` here? (alternatively, how come you are using it below if this one was making tests fail)

Comment (kaapstorm) [inline corehq/apps/users/models.py]: What I didn't understand until I tested it is that the data that is passed to wrap is loaded from JSON. So the error when trying to instantiate a DeviceIdLastUsed was hiding the error I got when trying to pass that DeviceIdLastUsed instance to `super().wrap()` (in line 819).

The one below, in the `create()` method works because it's building a Python object, and `commcare_user.created_on` has already been set to a datetime instance by `CouchUser.create()`.

File: corehq/apps/mobile_auth/views.py
Comment (proteusvacuum) [inline corehq/apps/mobile_auth/views.py]: would be nice to have a test for this too. Looks like that method will only work if the migration has been run on that specific user. Just making sure that this will always be the case.

File: corehq/apps/users/models.py
Comment (proteusvacuum) [inline corehq/apps/users/models.py]: I see. Thanks for the explanation, that makes sense.

File: corehq/apps/mobile_auth/views.py
Comment (czue) [inline corehq/apps/mobile_auth/views.py]: it should be since the migration is called automatically in the `wrap` function. though thinking about this a bit more I'm inclined to not deal with the migration and make a new property. will update this.

File: corehq/apps/mobile_auth/views.py
Comment (dimagimon) [inline corehq/apps/mobile_auth/views.py]: F401 'DeviceIdLastUsed' imported but unused

File: corehq/apps/users/tests/test_user_model.py
Comment (dimagimon) [inline corehq/apps/users/tests/test_user_model.py]: F401 'DeviceIdLastUsed' imported but unused

**Comment by czue** [comment]
updated but don't merge until i have a chance to sanity test

**Comment by czue** [comment]
ok this should be good to go. @benrudolph @proteusvacuum

**Comment by benrudolph** [comment]
(note if someone merges this before me please use squash on merge)

**Review by benrudolph — APPROVED** [review]

**Comment by benrudolph** [comment]
(killed second to last build)

**Pull request merged**
